### ISSUE TITLE
SC2317: false positive when using trap and capturing output

### ISSUE THREAD

**Issue by gpakosz** [koalaman/shellcheck#3219]
Hello,

```
$ shellcheck --version
ShellCheck - shell script analysis tool
version: 0.10.0
license: GNU General Public License, version 3
website: https://www.shellcheck.net
```

The following snippet emits `SC2317`:
```
#!/usr/bin/env bash

foo() {
  bar() {
    baz() {
      trap - INT
    }
    trap baz INT
  }
  out=$(bar)
  echo "$out"
}

if [ -n "$BASH_VERSION" ] && [ "${BASH_SOURCE[0]}" = "$0" ]; then
  foo "$@"
fi
```

```
$ shellcheck /tmp/trap.sh

In /tmp/trap.sh line 6:
      trap - INT
      ^--------^ SC2317 (info): Command appears to be unreachable. Check usage (or ignore if invoked indirectly).

For more information:
  https://www.shellcheck.net/wiki/SC2317 -- Command appears to be unreachable...
```

However, if I don't capture the output of `bar()`, I don't get the warning:
```
#!/usr/bin/env bash

foo() {
  bar() {
    baz() {
      trap - INT
    }
    trap baz INT
  }
  bar
}

if [ -n "$BASH_VERSION" ] && [ "${BASH_SOURCE[0]}" = "$0" ]; then
  foo "$@"
fi
```

In both case, `foo()` is assumed to be executed and `foo()` calls `bar()`
